src/logfile_values_experimental.py

```diff
@@ -0,0 +1,244 @@
+#
+"""
+logfile_values.py -- a python script to find metrics values in log file
+
+This script can retrive last line values by column number:
+
+python logfile_values.py --metric "metric1:1" "metric2:2" ...
+
+And aggregate data by time period with different functions:
+
+python logfile_values.py  --metric "metric1:1:mean:1d" --metric "metric2:3:count:60s"
+
+"""
+import re
+import time
+import os
+import gzip
+import logging
+
+from argparse import ArgumentParser, ArgumentTypeError
+from datetime import timedelta
+
+
+class Metric:
+    def __init__(self, arg):
+        if ':' not in arg:
+            raise ArgumentTypeError('Argument must have ":"')
+        parts = arg.split(':')
+        if len(parts) != 4 and len(parts) != 2:
+            raise ValueError('Wrong number of options')
+
+        # Accepts 2 or 4 options. If 2 then function and period are NULL
+        parts += [None] * (4 - len(parts))
+        self.name, column, function, period = parts
+        if period:
+            pattern = re.compile("^\d+[A-Za-z]{1,3}$")
+            if not pattern.match(period):
+                raise ArgumentTypeError('Period must have number and unit')
+        if not column.isdecimal():
+            raise ArgumentTypeError('Column must be a number')
+        self.column = int(column)
+        self.function = function
+        self.period = period
+        # Here is container for metric values
+        self.values = []
+        self.last_value = 0
+        self.now = int(time.time())
+
+    def get_timeshift(self):
+        if self.period:
+            value = int(self.period[:-1])
+            unit = self.period[-1].lower()
+            if unit == 's':
+                return timedelta(seconds=value).total_seconds()
+            if unit == 'min':
+                return timedelta(minutes=value).total_seconds()
+            if unit == 'h':
+                return timedelta(hours=value).total_seconds()
+            if unit == 'd':
+                return timedelta(days=value).total_seconds()
+        else:
+            return 0
+
+    def get_median(self):
+        sorted_list = sorted(self.values)
+        i = len(sorted_list)
+        if not i % 2:
+            return (sorted_list[(i // 2) - 1] + sorted_list[i // 2]) / 2
+        return sorted_list[i // 2]
+
+    def get_sum(self):
+        return sum(self.values)
+
+    def get_count(self, v=0):
+        return sum(1 for x in self.values if x >= v)
+
+    def get_mean(self):
+        return sum(self.values) / len(self.values)
+
+    def get_min(self):
+        return min(self.values)
+
+    def get_max(self):
+        return max(self.values)
+
+    def get_last_value(self):
+        return self.last_value
+
+    def get_metric_value(self):
+        if self.function:
+            if 'count_' in self.function:
+                return int(self.get_count(int(self.function.split('_')[1])))
+            return int(getattr(self, 'get_' + self.function)())
+        else:
```

Comment:
No need to `else:`.

Reply:
Need it, cuz idea is when user does not point function then we just return the last value for current metric

Reply:
You don't need the `else:` part because the previous line `return`s.